application/controllers/ParticipantController.php

```diff
@@ -199,6 +199,43 @@ public function allSchemesAction() {
         $this->view->globalQcAccess=$commonService->getConfig('qc_access');
     }
 
+    public function surveyAction() {
+        $this->_helper->layout()->activeMenu = 'view-reports';
+        $this->_helper->layout()->activeSubMenu = 'individual-reports';
+        if ($this->_hasParam('62cc026d0c2ab')) {
+            $db = Zend_Db_Table_Abstract::getDefaultAdapter();
+            $map_id = (int) base64_decode($this->_getParam('62cc026d0c2ab'));
+            if ($this->getRequest()->isPost()) {
+                $params = $this->_getAllParams();
+                $response = isset($params['response']) ? json_encode($params['response']) : null;
+                $result = $db->update('shipment_participant_map', [
+                    'cs_survey_response' => empty($response) ? null : $response
+                ], sprintf('map_id = %u', $map_id));
+                $this->_redirect("/participant/report");
```

Comment:
Will this prevent further execution of the code inside this function or should you add a return statement here?

Reply:
This will exit upon _redirect, unless you change $this->_exit = false